### ISSUE TITLE
3D Array does not work for conjugate sampler.

### ISSUE THREAD

**Issue by ethan-alt** [nimble-dev/nimble#1194]
I wanted to conduct a mixture of multivariate normals problem. The code I have is as follows:
```{r}
library(nimble)
## NIMBLE CODE--PRODUCES AN ERROR
code.multi <- nimbleCode({
  gamma ~ dbeta(gamma_shape1, gamma_shape2)
  for ( i in 1:2 ) {
    mu[1:J,i] ~ dmnorm(mu_mean[1:J], cov = mu_cov[1:J,1:J])
    Sigma[1:J,1:J,i] ~ dinvwish(S = S[1:J, 1:J], df = df)
  }
  for ( i in 1:n ) {
    c[i] ~ dbinom(size = 1, prob = gamma)
    d[i] <- c[i] + 1
    y[1:J,i] ~ dmnorm(mu[1:J, d[i]], cov = Sigma[1:J,1:J,d[i]] )
  }
})

## TEST THE CODE
n = 100
constants <- list(
    'n' = n
  , 'J' = 2
  , 'gamma_shape1' = 2
  , 'gamma_shape2' = 2
  , 'mu_mean'      = rep(0, 2)
  , 'mu_cov'       = diag(100, 2)
  , 'S'            = (2 + 10) * diag(2)
  , 'df'           = 2 + 10
)
inits <- list(
    'mu'    = cbind(c(-1, -1), c(1, 1))
  , 'c'     = rbinom(n = constants$n, size = 1, prob = 0.5)
  , 'gamma' = 0.5
  , 'Sigma' = abind( diag(2, 2), diag(1, 2), along = 3 )
)
data <- list('y' = t(mvtnorm::rmvnorm(constants$n, sigma = diag(2))) )
sample <- nimbleMCMC(
  code = code.multi, constants = constants, data = data, inits = inits, thin = 1, niter = 2000, nburnin = 1000
)
```

The sampler threw the following error:
```{r}
Defining model
Building model
Setting data and initial values
Running calculate on model
  [Note] Any error reports that follow may simply reflect missing values in model variables.
Checking model sizes and dimensions
Checking model calculations
Compiling
  [Note] This may take a minute.
  [Note] Use 'showCompilerOutput = TRUE' to see C++ compilation details.
Error: Error, wrong number of indices provided for dep_dmnorm_identity_coeff[iDep].
 This occurred for: dep_dmnorm_identity_coeff[iDep]
 This was part of the call:  dep_dmnorm_identity_coeff[iDep] <<- Interm_4091[1, 1]
```

Strangely, it seems to work when a non-conjugate sampler is used for the covariance matrices.

If there is a workaround (that retains conjugacy) please let me know!

**Comment by paciorek** [comment]
With conjugate samplers we do some processing to try to reduce unnecessary calculations. Looks like there is a bug in dealing with some indexing/dimensionality for the `dmnorm` case. We're likely to have a bug fix release shortly, so I'll try to look at this soon to see if we can fix for that.

**Comment by paciorek** [comment]
The issue is in the sampler for the `dwish` / `dinvwish` case. We special case the assignment of the dimension of `dep_dmnorm_multiplicativeScalar_coeff` based on knowing that we only handle scalar multipliers of the target node in the var or cov parameter of the dmnorm dependency. 

The special case handles the standard situation but mistakenly leaves out checking if we were in the dynamic dependencies case.
Need to add that logic to line 582 of `MCMC_conjugacy.R` so that dimension of `dep_dmnorm_identity_coeff` is also 1-d when we have dynamic indexing with `dwish` / `dinvwish` target.

We don't have full testing of the dynamic dependency + dmnorm/dwish/dinvwish cases - should add tests that try to do compilation (which would have caught this issue) and that check that we are correctly dynamically determining which potential dependents are actually dependents given the current value of the dynamic indices by checking for zeroes in `dep_dmnorm_<type>_coeff`.

**Comment by paciorek** [comment]
Fixed by PR #1200.
